tkg/client/cluster.go

```diff
@@ -199,12 +199,17 @@ func (c *TkgClient) CreateCluster(options *CreateClusterOptions, waitForCluster
 }
 
 // getClusterConfigurationBytes returns cluster configuration by taking into consideration of legacy vs clusterclass based cluster creation
-func (c *TkgClient) getClusterConfigurationBytes(options *ClusterConfigOptions, infraProviderName string, isManagementCluster, isWindowsWorkloadCluster bool) ([]byte, error) {
+func (c *TkgClient) getClusterConfigurationBytes(options *ClusterConfigOptions, infraProviderName string, isManagementCluster, isWindowsWorkloadCluster, isGenerateOnly bool) ([]byte, error) {
 	deployClusterClassBasedCluster, err := c.ShouldDeployClusterClassBasedCluster(isManagementCluster)
 	if err != nil {
 		return nil, err
 	}
 
+	// if DRY_RUN_MODE is cc_capable, generate addon manifest under cc capable mode, will not enable ccbase
+	if isGenerateOnly && os.Getenv("DRY_RUN_MODE") == "cc_capable" && os.Getenv("FILTER_BY_ADDON_TYPE") != "" {
```

Comment:
Do not use hardcode string DRY_RUN_MODE". Use const like ConfigVariableDefaultBomFile